### PR TITLE
Fixes #1001 CannotBeMergedDueNotInMergeableState

### LINKED ISSUE #1001: Review CannotBeMergedDueNotInMergeableState test
When this test was first written, it was designed to be in a state which would raise an exception if the consumer tried to merge it. That's now not the case.

TODO:

 - tweak the commits on the branch to see that it is back in an unmergeable state
 - verify the test now fails with the appropriate exception

### PR DESCRIPTION
**Pull request by gabrielweyer**
Fixes #1001 

I made two changes:

 1. I updated `master` head with the new commit so that the PR is not mergeable (not sure if it's the best way to do this.
 2. The `Merge` method wasn't awaiting the API call so the exception couldn't be caught, I made the method `async` and I'm now awaiting the `PUT` request.

### CODE REVIEW

File: Octokit/Clients/PullRequestsClient.cs [hunk 1]
```diff
@@ -118,7 +118,7 @@ public Task<PullRequestMerge> Merge(string owner, string name, int number, Merge
 
             try
             {
-                return ApiConnection.Put<PullRequestMerge>(ApiUrls.MergePullRequest(owner, name, number), mergePullRequest);
+                return await ApiConnection.Put<PullRequestMerge>(ApiUrls.MergePullRequest(owner, name, number), mergePullRequest);
```
Comment (khellang) [inline Octokit/Clients/PullRequestsClient.cs]: `ConfigureAwait(false)`?

Comment (gabrielweyer) [inline Octokit/Clients/PullRequestsClient.cs]: :+1: pushed another commit fixing this.

**Comment by Haacked** [comment]
> I updated master head with the new commit so that the PR is not mergeable (not sure if it's the best way to do this.

Seems reasonable to me.

The funny part is that this branch is also not mergeable and I got confused for a second about what you were making not mergeable. :stuck_out_tongue: Please fix up this branch so we can merge it in. Thanks!

**Comment by shiftkey** [comment]
Thanks @gabrielweyer!

**Pull request merged**